specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,383 @@
+                    iterator.MoveNext();
+                }
+                cm.IsOpen = true;
+            }
+        }, null);
+    };
+```
+# Remarks
+
+# API Notes
+
+# API Details
+ ```cpp
+    interface ICoreWebView2_4;
+    interface ICoreWebView2ContextMenuItem;
+    interface ICoreWebView2ContextMenuItemCollection;
+    interface ICoreWebView2ContextMenuItemCollectionIterator;
+    interface ICoreWebView2ContextMenuRequestedEventArgs;
+    interface ICoreWebView2ContextMenuRequestedEventHandler;
+
+    /// Defines the Context Menu Items
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_ID {
+        /// Custome type, will include Items brought in by browser extensions
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_CUSTOM,
+
+        /// No Command
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_NONE,
+
+        /// Page Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_BACK,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_FORWARD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_RELOAD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_STOP,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_NEW_WINDOW,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_PRINT,
+
+        /// Clipboard Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_ID_CUT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_COPY,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_PASTE,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_PASTE_AS_PLAIN_TEXT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_SELECT_ALL,
+
+        /// Saving
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_SAVE_LINK_AS,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_SAVE_IMAGE_AS,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_INSPECT, 
+
+        ... /// list is not complete
+        
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_ID;
+    
+    /// Indicates the context selected (where the user right clicked)
+    [v1_enum]
+    typedef enum COREWEBVIEW2_SELECTION_CONTEXT {
+
+        /// Indicates that a user right clicked on an image.
+
+        COREWEBVIEW2_SELECTION_CONTEXT_IMAGE,
+
+        /// Indicates that a user right clicked on a link.
+
+        COREWEBVIEW2_SELECTION_CONTEXT_LINK,
+
+        /// Indicates that a user right clicked on a textbox.
+
+        COREWEBVIEW2_SELECTION_CONTEXT_TEXTBOX,
+
+        /// Indicates that a user right clicked on selected text.
+
+        COREWEBVIEW2_SELECTION_CONTEXT_TEXT,
+
+        /// Indicates that a user right clicked on an area on the page.
+
+        COREWEBVIEW2_SELECTION_CONTEXT_PAGE,
+
+        /// Indicates that a user right clicked on a video
+
+        COREWEBVIEW2_SELECTION_CONTEXT_VIDEO,
+        
+        /// Indicates that a user right clicked on a pdf
+
+        COREWEBVIEW2_SELECTION_CONTEXT_PDF,
+
+        /// list is not complete
+
+    } COREWEBVIEW2_SELECTION_CONTEXT;
+
+    /// Context Menu Items displayed by the Edge browser, holds the properties of a context menu item
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown {
```

Comment:
Some context menu items have icons, checkboxes or radio buttons: 
![image](https://user-images.githubusercontent.com/2541336/121429488-d50b5b00-c92b-11eb-968c-1d0704fe3c3e.png)

I'm just looking at the browser default context menu. If you look at the browser extension's context menu API, are there more properties associated with context menu items to include?

Reply:
How about the icons on menu items?